plugins/output-elasticsearch/src/main/scala/com/stratio/sparkta/plugin/output/elasticsearch/ElasticSearchOutput.scala

```diff
@@ -47,74 +47,66 @@ class ElasticSearchOutput(keyName: String,
+  def getSchema: Seq[TableSchema] = bcSchema.get.filter(_.outputName == keyName).map(getTableSchemaFixedId)
+
+  private def createIndexAccordingToSchema(tableName: String, schema: StructType) =
     elasticClient.execute {
       create index tableName shards 1 replicas 0 mappings (
         mappingType.get.toLowerCase as (getElasticsearchFields(schema))
         )
     }
-  }
 
-  override def doPersist(stream: DStream[(DimensionValuesTime, Map[String, Option[Any]])]): Unit = {
+  override def doPersist(stream: DStream[(DimensionValuesTime, Map[String, Option[Any]])]): Unit =
     persistDataFrame(stream)
-  }
 
   override def upsert(dataFrame: DataFrame, tableName: String, timeDimension: String): Unit = {
     val sparkConfig = getSparkConfig(timeDimension, idField.isDefined || isAutoCalculateId)
-    val indexNameType = (tableName + "/" + mappingType.get).toLowerCase
-    dataFrame.saveToEs(indexNameType, sparkConfig)
+    dataFrame.saveToEs(indexNameType(tableName), sparkConfig)
   }
 
   //scalastyle:off
   def getElasticsearchFields(schema: StructType): Seq[TypedFieldDefinition] = {
     schema.map(structField => structField.dataType match {
-      case org.apache.spark.sql.types.LongType => structField.name typed FieldType.LongType
-      case org.apache.spark.sql.types.DoubleType => structField.name typed FieldType.DoubleType
-      case org.apache.spark.sql.types.DecimalType() => structField.name typed FieldType.DoubleType
-      case org.apache.spark.sql.types.IntegerType => structField.name typed FieldType.IntegerType
-      case org.apache.spark.sql.types.BooleanType => structField.name typed FieldType.BooleanType
-      case org.apache.spark.sql.types.DateType => structField.name typed FieldType.DateType
-      case org.apache.spark.sql.types.TimestampType => structField.name typed FieldType.DateType
-      case org.apache.spark.sql.types.ArrayType(_, _) => structField.name typed FieldType.MultiFieldType
-      case org.apache.spark.sql.types.MapType(_, _, _) => structField.name typed FieldType.ObjectType
-      case org.apache.spark.sql.types.StringType => structField.name typed FieldType.StringType index "not_analyzed"
+      case LongType => structField.name typed FieldType.LongType
+      case DoubleType => structField.name typed FieldType.DoubleType
+      case DecimalType() => structField.name typed FieldType.DoubleType
+      case IntegerType => structField.name typed FieldType.IntegerType
+      case BooleanType => structField.name typed FieldType.BooleanType
+      case DateType => structField.name typed FieldType.DateType
+      case TimestampType => structField.name typed FieldType.DateType
+      case ArrayType(_, _) => structField.name typed FieldType.MultiFieldType
+      case MapType(_, _, _) => structField.name typed FieldType.ObjectType
+      case StringType => structField.name typed FieldType.StringType index "not_analyzed"
       case _ => structField.name typed FieldType.BinaryType
     })
   }
```

Comment:
Mongo format is something like ```"host:port,host:port"```. We can change ```MongoDbOutput.getConnentionConfs``` to something like this:

```
 override val nodes = properties.getHostPortConfs("nodes", DefaultNode, DefaultPort)

// Input (Seq(host,port),(host,port)), output "host:port,host:port"
val nodesFormatted = nodes.map(t => s"${t._1}:${t._2}").mkString(",")
```
What do you think?